### ISSUE TITLE
Storage of already-processed posts

### ISSUE THREAD

**Issue by dixtergordong** [Vitards/Luca-Brasi-Bot#6]
We'll need a way to keep track of what posts the bot has already looked at (in the event that the server gets restarted, for example). I would like not to introduce a database since this _should_ be a simple app, so my first instinct is just to throw it in a JSON file. Thoughts?

**Comment by nb311** [comment]
I agree with this. Will still need to store that somewhere where we can read and update as we go. I want to say S3 but I'm not too keen on us managing an AWS account. What kind of storage solution did you have in mind for the file?

Edit: If we go with Heroku, it might be easier for us to use their free tier postgres solution for this. Although you mentioned you want to stay away from a database, I like that this is a free option available to us. Also, I don't think this should be too much of a headache at all and would be happy to set this up.

**Comment by dixtergordong** [comment]
Oh nice, i was unaware that heroku has a free tier postgres, that’s awesome.

As a first iteration, we could have a Posts table which basically just stores the id. I’m not sure we actually need anything other than that, at least at first. Just brainstorming here but we could also cache market cap and `isSPAC` for tickets as well (since i don’t think we always need to get the absolute latest Market Cap and info for a ticker... imagine how often we’d be making API calls for $MT and $CLF lol).

Having an ORM is probably overkill at this point, but I wouldn’t be against it anyway since it’s easy enough and is more flexible for more features down the road. I’ve had nice success with TypeORM. Lemme know your thoughts @nb311 and @boyan-soubachov

**Comment by boyan-soubachov** [comment]
I’d avoid an ORM personally for something this simple. If possible, I’d avoid saving state altogether (i.e. no DB). 

Would it make sense if we first built it as a stateless service without a DB and should we find that we need it, we can add it later?

I agree that is making excessive calls to the Finance API _could_ be an issue but I think even if we just cache those in memory, we might get away with it just fine and not have to deal with the extra time, costs, and headaches associated with maintaining a DB.
